bin/check-azurerm-monitor-metric.rb

```diff
@@ -0,0 +1,305 @@
+  option :resource_id,
+         description:  'The full id of the resource.  If given, the resource namespace/type/group along with subscription id are ignored.',
+         short: '-r ID',
+         long: '--resource-id ID',
+         default: ''
+
+  option :metric,
+         description:  'The name of the metric',
+         short: '-m ID',
+         long: '--metric ID',
+         required: true
+
+  option :filter,
+         description:  "The filter applied to the metric.  See Azure docs for the syntax.  Note: This can be used to segment the return by dimensions, so that the script checks each dimension separately. eg APIName eq '*'",
+         short: '-f FILTER',
+         long: '--filter FILTER'
+
+  option :aggregation,
+         description:  'Aggregation.  This can be Average, Count, Maximum, Minimum, Total',
+         short: '-a aggregation',
+         long: '--aggregation aggregation',
+         default: 'average'
+
+  option :warning_over,
+         description: 'The warning threshold to check if the metric is forecasted to go over.',
+         short: '-w N',
+         long: '--warning WARN',
+         proc: proc { |val| val.to_i }
+
+  option :critical_over,
+         description: 'The critical threshold to check if the metric is forecasted to go over.',
+         short: '-c N',
+         long: '--critical CRIT',
+         proc: proc { |val| val.to_i }
+
+  option :warning_under,
+         description: 'The warning threshold to check if the metric is forecasted to go under.',
+         short: '-W N',
+         long: '--warning-under WARN',
+         proc: proc { |val| val.to_i }
+
+  option :critical_under,
+         description: 'The critical threshold to check if the metric is forecasted to go under.',
+         short: '-C N',
+         long: '--critical-under CRIT',
+         proc: proc { |val| val.to_i }
+
+  def run
+    if config[:resource_id].to_s.empty? && config[:resource_name].to_s.empty?
+      unknown 'resource id or resource name/group/type/namespece and subscription id must be provided'
+    end
+
+    if !config[:critical_over] && !config[:warning_over] && !config[:critical_under] && !config[:warning_under]
+      unknown 'At least one threshold must be provided.'
+    end
+
+    if last_metric_values.empty?
+      unknown "There are no metric values for #{config[:metric]} on resource #{config[:resource_id] || config[:resource_name]} with aggregation #{config[:aggregation]}"
+    else
+      last_metric_values.each do |metric_val|
+        if config[:critical_over] && metric_val[:value] > config[:critical_over].to_f
+          critical "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        elsif config[:warning_over] && metric_val[:value] > config[:warning_over].to_f
+          warning "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        elsif config[:critical_under] && metric_val[:value] < config[:critical_under].to_f
```

Comment:
So you could not say specify a utilization range then? I think there are use cases where you want things to stay in a range. Queued worker messages for non backend processing work comes to mind. Network utilization from a NAT comes to mind, if there is ever under a certain amount of traffic there is something horribly wrong if its too high you need to look into scale your NAT throughput.

I am not asking you to add it just talking through some things that someone might want in the future. You might find you have use cases you did not think about and want this even if its in another PR.

If we wanted to do this we could just keep track of the critical, warnings for each and at the end we could include all crit and/or warning

Reply:
I'll try to fit this change in.